Expensify/App
# Move Search to a RightDockedModal

@marcaaron 

### Details
Change the search from a sidebar component to a `RightDockedModal` component.
As a result, got rid of everything related to `ChatSwitcher`, since the Sidebar no longer has it.

#### Problem I am seeing (not new, but important to point out)

I have been seeing that warning
```
Warning: Can't perform a React state update on an unmounted component (...)
```
since the first issues I have worked on, and it can be seen on the console every now and then when clicking `New Chat` or `New Group`. For the search, it seems to be consistently showing up every first time a contact is selected from the search list.

I find it a hard warning to debug since the message is not very specific, but I am going to give it a try.


### Fixed Issues
Fixes #1448 

### Tests
1. Click on the Magnifying glass icon in the sidebar;
2. If on small screens, check if the `SearchPage` covers all the screen. If on large screens, check if the `SearchPage` is a modal that animates from the right, just as the `NewChatPage` and `NewGroupPage` do;
3. Verify that when selecting a contact, the selected report chat opens up, closing the `SearchPage`;
4. Verify if the modal closes either by clicking anywhere outside of it, or by pressing `ESC`, or by pressing / clicking on the `X` button on the top right;
5. **Edit (adding)** Also, verify that the other keyboard actions work accordingly (`UP` / `DOWN` arrow keys to change selection, `ENTER` to perform the selection.

### Tested On


### Screenshots

#### Web
![web](https://i.ibb.co/X3gBtGn/web.gif)

#### Mobile Web
![mobile web](https://i.ibb.co/h2fRgxN/mobile-web.gif)

#### Desktop
![desktop](https://i.ibb.co/HxMRs3D/desktop.gif)

#### iOS
![ios](https://i.ibb.co/xfbMp7s/ios.png)

#### Android
![android](https://i.ibb.co/37GB7M9/android.gif)
